connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -525,6 +525,19 @@ class SparkSession private[sql] (
     planIdGenerator.set(0)
   }
 
+  /**
+   * Interrupt all operations of this session currently running on the connected server.
+   *
+   * TODO/WIP: Currently it will interrupt the Spark Jobs running on the server, triggered from
+   * ExecutePlan requests. If an operation is not running a Spark Job, it becomes an noop and the
```

Comment:
Is there an example that an operation does not run a spark job now but later run more jobs, thus the interrupt all  API does not work with it?

Reply:
For example if a query just started and is still doing analysis/optimization/planning when the interrupt hits, the interrupt will be a noop and the query will still launch jobs once it gets to execution.
For example, Delta DML commands often run several stages, each of them runs jobs, with some driver local processing in between.

Reply:
When I move execution to a separate thread, I can `Thread.interrupt` it and then it should be checked any time it gets blocked/waiting + may need some explicit checks for `Thread.isInterrupted()` in between.